azurerm/internal/services/iothub/iothub_route_resource.go

```diff
@@ -57,6 +57,7 @@ func resourceIotHubRoute() *pluginsdk.Resource {
 				Type:     pluginsdk.TypeString,
 				Required: true,
 				ValidateFunc: validation.StringInSlice([]string{
+					string(devices.RoutingSourceDeviceConnectionStateEvents),
```

Comment:
```suggestion
					"DeviceConnectionStateEvents" //TODO update with enum when added to sdk
```

For now, we could add this as a string and include a TODO note with the link to the sdk issue so it can be updated later.

Reply:
Added in commit 1638f2643061efb62c13dff5f409ffb40e2aa68e.